src/movies.js

```diff
@@ -1,31 +1,112 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require('./data');
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  const directors = array.map(function (movie) {
+    return movie.director;
+  });
+  return directors.filter(function (director, i) {
+    return directors.indexOf(director) === i;
+  });
+}
+
+// BONUS 1 FILTER
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(array) {
+  const movieTotal = array.filter(function (movie) {
+    if (
+      movie.director === 'Steven Spielberg' &&
+      movie.genre.includes('Drama')
+    ) {
+      return true;
+    }
+  });
+  return movieTotal.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
-
+function scoresAverage(array) {
+  const movieScore = array.reduce(function (accummulator, currentValue) {
+    if (array.score === '' || array.score === undefined) {
+      array.score = 0;
+    }
+    return accummulator + currentValue.score;
+  }, 0);
+  if (!array.length) {
+    return 0;
+  }
+  let averageMovie = movieScore / array.length;
+  return +averageMovie.toFixed(2);
+}
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(array) {
+  let dramaMovies = array.filter(function (movie) {
+    if (movie.genre.includes('Drama')) {
+      return true;
+    }
+  });
+  let dramaTotal = dramaMovies.reduce(function (accummulator, currentValue) {
+    return accummulator + currentValue.score;
+  }, 0);
+  if (!dramaMovies.length) {
+    return 0;
+  }
+  let dramaAverage = dramaTotal / dramaMovies.length;
+  return +dramaAverage.toFixed(2);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(array) {
+  // let sortedMovies = [];
+
+  return [...array].sort(function (a, b) {
+    if (a.year === b.year) {
+      return a.title.localeCompare(b.title);
+    }
+    return a.year - b.year;
+  });
+  // sortedMovies.push(dateOrder);
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(array) {
+  let dateOrder = [...array].sort(function (a, b) {
+    if (a.title < b.title) {
+      return -1;
+    } else if (a.title > b.title) {
+      return 1;
+    }
+  });
+  return dateOrder.slice(0, 20);
+}
```

Comment:
We need to use localeCompare again, as this won't sort everything right. 

.sort( (a,b) =>  a.localeCompare(b) )